lib/run-compilation.js

```diff
@@ -26,19 +27,12 @@ module.exports = function runCompilation (options, compiler, done) {
         errors = results.filter(fileHasErrors);
       }
 
-      if (options.quiet === false) {
-        console.warn(options.formatter(results));
-      }
-
-      if (options.failOnError && errors.length) {
-        done(new Error(errorMessage));
-      } else {
-        done();
-      }
-    })
-    .catch(done);
+      if (options.quiet === false) console.warn(options.formatter(results));
+      if (options.failOnError && errors.length) done(new Error(errorMessage));
+      else done();
```

Comment:
No problem at all => added a new commit to address these. I also added another commit for possibly changing the readme description.